### PR TITLE
Complete `init` functionality

### PR DESCRIPTION
**Pull request by Nazeeh21**
Add `init` functionality in `/packages/dapp/src/pages/index.tsx`

### CODE REVIEW

File: packages/dapp/helpers/index.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+export const getDidFromTokenURI = tokenURI => {
+  const [ipfsWithColon, _, cid, didFilename] = tokenURI.split("/");
+  const [did] = didFilename.split(".json");
+  return {
+    did,
+    cid,
+    tokenURI,
+    filename: didFilename,
+  };
+};
```
[the remark this answers is not in the document]
Comment (Nazeeh21) [reply packages/dapp/helpers/index.js:10]: on it

File: packages/dapp/src/pages/index.tsx [hunk 2]
```diff
@@ -24,6 +43,60 @@ const Home = () => {
...
   };
 
+  const init = async () => {
+    if (library && library.getSigner() && contracts) {
+      try {
+        const signer = library.getSigner();
+        const tokenAddress = await contracts.yourReadContract.token();
+        console.log(`tokenAddress`, tokenAddress);
+
+        const tokenContract = new ethers.Contract(
```
Comment (damianmarti) [inline packages/dapp/src/pages/index.tsx:53]: Maybe we can add it on Web3Provider after initialize the DRecruitV1 contract and change yourReadContract to yourReadContracts (the same for write), and be able to use the token contract from the context.

Comment (Nazeeh21) [reply packages/dapp/src/pages/index.tsx:53]: This `contracts` variable from the `Web3Context` consists of DRecruit contract. So I'll add new variable maybe `tokenContract` to `Web3Context` for our token contract

**Review by damianmarti — APPROVED** [review]

**Comment by Nazeeh21** [comment]
I've stored the tokenAddress contract in `Web3Context`. Please have a look and provide suggestions

**Comment by damianmarti** [comment]
> I've stored the tokenAddress contract in `Web3Context`. Please have a look and provide suggestions

LGTM. 

We will need to change in the read contracts to use the static provider, but we can change it later, so we can merge this PR to be able to work on the other pages.

**Comment by Nazeeh21** [comment]
@Dhaiwat10 please have a look. I think it is ready to get merged.

**Review by Cali93 — APPROVED** [review]
LGTM but we could remove the setTokenContracts and just use
`setContracts({ readDrecruitContract, writeDrecruitContract, readTokenContract, writeTokenContract })`

**Comment by Nazeeh21** [comment]
Okay, I'll do it that way

**Comment by Nazeeh21** [comment]
@Cali93 I've stored `tokenContract` in `contracts` variable. Please have a look and let me know if any changes are required

File: packages/dapp/src/contexts/Web3Provider.tsx [hunk 3]
```diff
@@ -80,6 +81,13 @@ const Web3Provider = ({ children }: { children: any }) => {
     });
   };
 
+  const setTokenContract = (contract: null | any) => {
```
Comment (Cali93) [inline packages/dapp/src/contexts/Web3Provider.tsx:84]: You can remove this block now that we're just using setContracts

**Review by Cali93 — APPROVED** [review]

File: packages/dapp/src/contexts/Web3Provider.tsx (hunk 3 above)
Comment (Nazeeh21) [reply packages/dapp/src/contexts/Web3Provider.tsx]: Removed.

**Pull request merged**